src/main/kotlin/com/example/moiza/domain/portfolio/domain/repository/PortfolioRepositoryImpl.kt

```diff
@@ -0,0 +1,41 @@
+package com.example.moiza.domain.portfolio.domain.repository
+
+import com.example.moiza.domain.portfolio.domain.QPortfolio.portfolio
+import com.example.moiza.domain.portfolio.domain.type.UserStatus
+import com.example.moiza.domain.portfolio.presentation.dto.res.PortfolioListResponse
+import com.querydsl.core.types.Projections.constructor
+import com.querydsl.jpa.impl.JPAQuery
+import com.querydsl.jpa.impl.JPAQueryFactory
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.data.support.PageableExecutionUtils
+import org.springframework.stereotype.Repository
+
+@Repository
+class PortfolioRepositoryImpl(
+    private val queryFactory: JPAQueryFactory
+): PortfolioRepositoryCustom {
+    override fun getPortfolioList(status: UserStatus, pageable: Pageable): Page<PortfolioListResponse> {
+        val fetch = queryFactory
+            .select(
+                constructor(
+                    PortfolioListResponse::class.java,
+                    portfolio.id,
+                    portfolio.user.nickname,
+                    portfolio.user.school,
+                    portfolio.user.major,
+                    portfolio.user.profile
+                )
+            )
+            .from(portfolio)
+            .where(portfolio.userStatus.loe(status.level))
+            .fetch()
+
```

Comment:
현재 fetch에서 pageable 정보를 활용하고 있지 않아서 전체 결과가 반환돼요.
```suggestion
val fetch = queryFactory
    .select(
        constructor(
            PortfolioListResponse::class.java,
            portfolio.id,
            portfolio.user.nickname,
            portfolio.user.school,
            portfolio.user.major,
            portfolio.user.profile
        )
    )
    .from(portfolio)
    .where(portfolio.userStatus.loe(status.level))
    .offset(pageable.offset)
    .limit(pageable.pageSize)
    .fetch()
```
이런식으로 적용하는건 어떨까요?

Reply:
`PageableExecutionUtils.getPage(fetch, pageable) { countQuery.fetchOne() ?: 0L }`
에서 pageable을 사용하고 있습니다

관련된 레퍼런스입니다.
https://rachel0115.tistory.com/entry/QueryDsl-Page-Slice-%ED%8E%98%EC%9D%B4%EC%A7%80%EB%84%A4%EC%9D%B4%EC%85%98-%EB%AC%B4%ED%95%9C-%EC%8A%A4%ED%81%AC%EB%A1%A4

https://nomoreft.tistory.com/45